src/System.Device.Gpio/Interop/Unix/libgpiod/V1/SafeLineHandle.cs

```diff
@@ -7,31 +7,53 @@
 namespace System.Device.Gpio.Libgpiod.V1;
 
 /// <summary>
-/// Pointer to a pin.
+/// Pointer to a pin (Not a real SafeLineHandle, because we need to align its finalization with the owning object)
 /// </summary>
-internal class SafeLineHandle : SafeHandle
+internal sealed class SafeLineHandle : IDisposable
 {
-    public PinMode PinMode { get; set; }
-
-    public SafeLineHandle()
-        : base(IntPtr.Zero, true)
+    private IntPtr _handle;
+    public SafeLineHandle(IntPtr handle)
     {
+        _handle = handle;
+        PinMode = PinMode.Input;
     }
 
-    protected override bool ReleaseHandle()
+    public PinMode PinMode { get; set; }
+
+    public IntPtr Handle
     {
-        // Contrary to intuition, this does not invalidate the handle (see comment on declaration)
-        LibgpiodV1.gpiod_line_release(handle);
-        return true;
+        get
+        {
+            if (_handle == IntPtr.Zero)
+            {
+                throw new ObjectDisposedException(nameof(SafeLineHandle));
+            }
+
+            return _handle;
+        }
+        set
+        {
+            _handle = value;
+        }
     }
 
     /// <summary>
-    /// Release the lock on the line handle. <see cref="LibgpiodV1.gpiod_line_release"/>
+    /// Release the lock on the line handle. <see cref="Interop.LibgpiodV1.gpiod_line_release"/>
     /// </summary>
     public void ReleaseLock()
     {
-        ReleaseHandle();
+        // Contrary to intuition, this does not invalidate the handle (see comment on declaration)
+        Interop.LibgpiodV1.gpiod_line_release(_handle);
```

Comment:
Assuming this will make it so that `IsInvalid` method will return true? Does it set the handle to IntPtr.Zero or to InvalidHandleValue?

Reply:
No, see comment. This just internally decreases some kind of reference counter and does not invalidate the handle. IIRC this is needed to change the mode of a pin without closing it.